operator/helper/operatortest/operatortest.go

```diff
@@ -0,0 +1,87 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package operatortest
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+	"testing"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
+
+	"github.com/open-telemetry/opentelemetry-log-collection/operator/helper"
+)
+
+// ConfigTestCase is used for testing golden configs
+type ConfigTestCase struct {
+	Name      string
+	Expect    interface{}
+	ExpectErr bool
+}
+
+func configFromFileViaYaml(file string, config interface{}) (interface{}, error) {
```

Comment:
Since both `yaml` and `mapstructure` both unmarshal directly into an interface, let's just mimic their behavior here as well.

Specifically, I believe you can change this to `func configFromFileViaYaml(file string, config interface{}) error {` and then only return nil or error.